sm/tedge_agent/src/agent.rs

```diff
@@ -113,15 +117,28 @@ pub struct SmAgent {
     config: SmAgentConfig,
     name: String,
     persistance_store: AgentStateRepository,
+    _flock: Flockfile,
+}
+
+pub async fn create_sm_agent<'a>(
+    app_name: &'a str,
+    sm_agent_config: SmAgentConfig,
+) -> Result<SmAgent, anyhow::Error> {
+    let _flock = check_another_instance_is_not_running(app_name)?;
+
+    info!("{} starting", app_name);
+
+    Ok(SmAgent::new(app_name, sm_agent_config, _flock))
```

Comment:
I wonder if this can be moved into `SmAgent::new` or `SmAgent::create`. The point is to use the struct as a scope.